source/features/repo-age.tsx

```diff
@@ -4,50 +4,87 @@ import cache from 'webext-storage-cache';
 import select from 'select-dom';
 import repoIcon from 'octicon/repo.svg';
 import elementReady from 'element-ready';
+import fetchDom from '../libs/fetch-dom';
 import features from '../libs/features';
-import * as api from '../libs/api';
-import {getRepoGQL, getRepoURL} from '../libs/utils';
+import {getCleanPathname, getRepoURL} from '../libs/utils';
 
 const dateFormatter = new Intl.DateTimeFormat('en-US', {
 	year: 'numeric',
 	month: 'long',
 	day: 'numeric'
 });
 
-const getRepoCreationDate = cache.function(async (): Promise<string> => {
-	const {repository} = await api.v4(`
-		repository(${getRepoGQL()}) {
-			createdAt
-		}
-	`);
+type OldestCommitDetails = {
+	url: string;
+	datetime: string;
+};
 
-	return repository.createdAt;
+const getOldestCommitDetails = cache.function(async (): Promise<OldestCommitDetails | false> => {
+	await elementReady<HTMLAnchorElement>('.commit-tease-sha', {target: select('.commit-tease')!});
+	const commitsCount = Number(select('li.commits .num')!.textContent!.replace(',', ''));
+	const lastCommitHash = select<HTMLAnchorElement>('.commit-tease-sha')!.href.split('/').pop()!;
```

Comment:
@fregante Even though I had added `await elementReady<HTMLAnchorElement>('.commit-tease-sha')` on Line 23, this was failing with error `Cannot read property href of null`. I added `target: select('.commit-tease')` and after that I did not see that error. Is this a bug?  

@yakov116 Can you also check if you are still getting an error?

Reply:
I am not, but... i cleared the cache so...

Reply:
`elementReady` may also return `null` because by default it stops at domready.

You can use `lastCommit = await elementReady(...)` to just get the item on the spot instead of using `select` again.

Also if that `target` is available on the page when `elementReady` runs, then you don't need `elementReady` because `commit-tease` and `commit-tease-sha` loaded together, I think.

Reply:
Got it. I passed `stopOnDomReady: false` and it's no longer causing an error.